moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h

```diff
@@ -116,7 +117,7 @@ class RobotTrajectory
 
   const moveit::core::RobotState& getWayPoint(std::size_t index) const
   {
-    return *waypoints_[index];
+    return *waypoints_.at(index);
```

Comment:
This change seems unrelated to using `rclcpp::Duration` and deserves its own conversation in a separate issue or pull request. Once we widen the API in this way, we cannot narrow it again without breaking the API so we must be very careful about such changes.